gnmi_subscribe/gnmi_subscribe.go

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2020 Google Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"strings"
+	"time"
+
+	log "github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/gnxi/utils"
+	"github.com/google/gnxi/utils/credentials"
+	"github.com/google/gnxi/utils/xpath"
+	pb "github.com/openconfig/gnmi/proto/gnmi"
+	"google.golang.org/grpc"
+)
+
+type arrayFlags []string
+
+func (i *arrayFlags) String() string {
+	return ""
+}
+
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+const (
+	defaultTimeout = 10
+)
+
+var (
+	subscribeClient    pb.GNMI_SubscribeClient
+	xPathFlags         arrayFlags
+	pbPathFlags        arrayFlags
+	targetAddr         = flag.String("target_addr", ":9339", "The target address in the format of host:port")
+	targetName         = flag.String("target_name", "", "The target name used to verify the hostname returned by TLS handshake")
+	connectionTimeout  = flag.Duration("timeout", defaultTimeout, "The timeout for a request, 10 seconds by default")
+	valueStreamingMode = flag.String("value_mode", "STREAM", "The mode of streaming values the target should use use, STREAM by default")
```

Comment:
I would define instead two booleans.
-once
-pool
 If none is mentioned then use stream (default). If both are defined, send error that only one of either can be used.

Reply:
That's a lot cleaner, thank you